src/handler/http/models/alerts/destinations.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright 2024 OpenObserve Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+use config::meta::alerts::destinations as meta_destinations;
+use hashbrown::HashMap;
+use serde::{Deserialize, Serialize};
+use utoipa::ToSchema;
+
+#[derive(Clone, Debug, Serialize, Deserialize, ToSchema)]
+pub struct Destination {
+    #[serde(default)]
+    pub name: String,
+    /// Required for `Http` destination_type
+    #[serde(default)]
+    pub url: String,
+    /// Required for `Http` destination_type
+    #[serde(default)]
+    pub method: HTTPType,
```

Comment:
This reflects the current behavior of the API. Removing `#[serde(default)]` would be a breaking change to the API since these would no longer be optional fields.